src/adapter/src/catalog/migrate.rs

```diff
@@ -7,29 +7,44 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use futures::future::BoxFuture;
+use mz_proto::RustType;
+use mz_storage_client::types::connections::ConnectionContext;
+use mz_storage_client::types::sources::{
+    PostgresSourcePublicationDetails, ProtoPostgresSourcePublicationDetails,
+};
+use prost::Message;
 use semver::Version;
 use std::collections::BTreeMap;
-use tracing::info;
+use tracing::{info, warn};
 
 use mz_ore::collections::CollectionExt;
 use mz_sql::ast::display::AstDisplay;
-use mz_sql::ast::{Raw, Statement, UnresolvedObjectName};
+use mz_sql::ast::{PgConfigOptionName, Raw, Statement, UnresolvedObjectName, Value};
 use mz_sql::plan::{Params, PlanContext};
 
 use crate::catalog::{Catalog, ConnCatalog, SerializedCatalogItem};
 
 use super::storage::Transaction;
 
-fn rewrite_items<F>(tx: &mut Transaction, mut f: F) -> Result<(), anyhow::Error>
+async fn rewrite_items<F>(
+    tx: &mut Transaction<'_>,
+    cat: Option<&ConnCatalog<'_>>,
```

Comment:
Maybe we should have two variants of `rewrite_items`, one with a `ConnCatalog` and one without. That way we don't have have this `Option` and `expect` below.

Alternatively, if we implement `Clone` for `ConnCatalog`, we don't need the `Option` and can just clone the `ConnCatalog` before passing it into the closure when we call this function.

Reply:
The migration code thrashes a lot, so I wanted to localize these changes as much as possible. I think that the aesthetics of doing something more robust are desirable but this code will be yanked out in two weeks so didn't want to spend any more energy on it than was necessary.

Reply:
Yeah good point, we have an issue open somewhere to re-work all of this anyways.